crates/matrix-sdk-base/src/store/migration_helpers.rs

```diff
@@ -253,3 +260,179 @@ impl From<RoomNameEventContentV1> for RoomNameEventContent {
         RoomNameEventContent::new(value.name.unwrap_or_default())
     }
 }
+
+/// [`DependentQueuedRequest`] v1.
+///
+/// See the docs of [`DependentQueuedRequestKindV1`] for details on the
+/// migration.
```

Comment:
Instead of duplicating `DependentQueuedRequestV1`, any chance we could deserialize `kind` from a new `SerializedDependentQueueRequestKind` struct, which would have a `serde(untagged)` annotation on top, and would try the latest format, and the previous v1 format if the latest didn't work? The v1 format only needs the previous format for the `FinishUpload` variant, not the others, if what I'm thinking about is correct.